osu.Game/Online/API/Requests/PatchBeatmapPackageRequest.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using osu.Framework.IO.Network;
+
+namespace osu.Game.Online.API.Requests
+{
+    public class PatchBeatmapPackageRequest : APIUploadRequest
+    {
+        protected override string Uri
+        {
+            get
+            {
+                // can be removed once the service has been successfully deployed to production
+                if (API!.Endpoints.BeatmapSubmissionServiceUrl == null)
+                    throw new NotSupportedException("Beatmap submission not supported in this configuration!");
+
+                return $@"{API!.Endpoints.BeatmapSubmissionServiceUrl!}/beatmapsets/{BeatmapSetID}";
+            }
+        }
+
+        protected override string Target => throw new NotSupportedException();
+
+        public uint BeatmapSetID { get; }
+
+        // ReSharper disable once CollectionNeverUpdated.Global
+        public Dictionary<string, byte[]> FilesChanged { get; } = new Dictionary<string, byte[]>();
+
+        // ReSharper disable once CollectionNeverUpdated.Global
```

Comment:
I do wonder if we'll remember to remove these

Reply:
They're going away in the next PR in series, no worries.